concordium-contracts-common/src/schema_json.rs

```diff
@@ -0,0 +1,728 @@
+use crate::{schema::*, *};
+//use anyhow::{anyhow, bail, ensure, Context};
+use num_bigint::{BigInt, BigUint};
+use num_traits::Zero;
+use serde_json::Value;
+use std::convert::TryInto;
+use thiserror;
+
+#[derive(Debug, thiserror::Error)]
+pub enum JsonError {
+    #[error("Failed writing")]
+    FailedWriting,
+    #[error("Unsigned integer required")]
+    UnsignedIntRequired,
+    #[error("Signed integer required")]
+    SignedIntRequired,
+    #[error("Failed parsing account address")]
+    FailedParsingAccountAddress,
+    #[error("{0}")]
+    WrongJsonType(String),
+    #[error("{0}")]
+    FieldError(String),
+    #[error("{0}")]
+    EnumError(String),
+    #[error("{0}")]
+    MapError(String),
+    #[error("{0}")]
+    PairError(String),
+    #[error("{0}")]
+    ArrayError(String),
+    #[error("{0}")]
+    ParseError(String),
+    #[error("{0}")]
+    ByteArrayError(String),
+    #[error("{0}")]
+    FromHexError(#[from] hex::FromHexError),
+    #[error("{0}")]
+    TryFromIntError(#[from] core::num::TryFromIntError),
+    #[error("{0}")]
+    ParseIntError(#[from] std::num::ParseIntError),
+    #[error("{0}")]
+    ParseDurationError(#[from] ParseDurationError),
+    #[error("{0}")]
+    ParseTimestampError(#[from] ParseTimestampError),
+}
+
+// Serializes a value to the provided output buffer
+macro_rules! serial {
+    ($value:ident, $out:ident) => {
+        $value.serial($out).or(Err(JsonError::FailedWriting))
+    };
+}
+
+// Returns a WrongJsonType error with the provided error message
+macro_rules! wrong_json_type {
+    ($msg:literal) => {
+        Err(JsonError::WrongJsonType($msg.to_string()))
+    };
+}
```

Comment:
Please remove this macro. THis does not bring value and just obfuscates the code.

Reply:
Use the constructor directly. The savings in a few characters are not worth it.

Reply:
Yeah looks better, thanks